From 6f03a2f937eb94d56bd99bc085c068113ea6e0de Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kamil=20Go=C5=82da?= <kamil.golda@protonmail.com>
Date: Sat, 18 Jan 2025 22:32:53 +0100
Subject: [PATCH 13/22] arm64: dts: qcom: sm6125: Add GPU clock controller
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Wire up the sm6125 GPU clock contoller in dts.

Signed-off-by: Kamil Gołda <kamil.golda@protonmail.com>
[fix reg]
Signed-off-by: Barnabás Czémán <barnabas.czeman@mainlining.org>
---
 arch/arm64/boot/dts/qcom/sm6125.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sm6125.dtsi b/arch/arm64/boot/dts/qcom/sm6125.dtsi
index 86de1560b..a67bb2be7 100644
--- a/arch/arm64/boot/dts/qcom/sm6125.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm6125.dtsi
@@ -1263,6 +1263,15 @@ sram@4690000 {
 			reg = <0x0 0x04690000 0x0 0x10000>;
 		};
 
+		gpucc: clock-controller@5990000 {
+			compatible = "qcom,sm6125-gpucc";
+			reg = <0x0 0x05990000 0x0 0x9000>;
+			clocks = <&rpmcc RPM_SMD_XO_CLK_SRC>,
+				 <&gcc GCC_GPU_GPLL0_CLK_SRC>;
+			#clock-cells = <1>;
+			#power-domain-cells = <1>;
+		};
+
 		mdss: display-subsystem@5e00000 {
 			compatible = "qcom,sm6125-mdss";
 			reg = <0x0 0x05e00000 0x0 0x1000>;
-- 
2.56.0

